### ISSUE TITLE
Problem with Signed Comparisons

### ISSUE THREAD

**Issue by DavePearce** [DavePearce/JEVM#2]
There appears to be a problem `Word.signedLessThan()`.  For example:

```
w256 w = new w256(0);
w256 v = new w256(5);
System.out.println(w + " < " + v + " = " + w.signedLessThan(v));
System.out.println(v + " < " + w + " = " + v.signedLessThan(w));
```

This returns:

```
0x0000000000000000000000000000000000000000000000000000000000000000 < 0x0000000000000000000000000000000000000000000000000000000000000005 = false
0x0000000000000000000000000000000000000000000000000000000000000005 < 0x0000000000000000000000000000000000000000000000000000000000000000 = true
```

**Comment by DavePearce** [comment]
Believe to be fixed now.

**Comment by dknz2008** [comment]
Problem (or similar problem) still exists. Executing -5 < 4 gives us the following bytecode
`STATE: OK;[0x0000000000000000000000000000000000000000000000000000000000000000];[];[];[PUSH1 0x05,PUSH1 0x00,SUB,PUSH1 0x04,SGT]`

However, we can see that the value 0 is on the stack, whilst it should be the the value 1, as

1 if µs[0] > µs[1]
0 otherwise

**Comment by DavePearce** [comment]
Hey Dylan,

Ok, I'm looking at this.  Hmmm, interesting.  Yes, I agree (after some thought).  It's actually a different bug though.  The order of execution for all the comparators (e.g. `LT`, `GT`, `SLT`, `SGT`) was the wrong way around I think.

**Comment by DavePearce** [comment]
Right, I have pushed through another fix and released it as version `0.1.6`.

**HOWEVER** I think you definitely need to test this and check its working as expected.  In particular, I've changed the order of execution for all the comparators.  It feels to me like that will break your code again ... though I could be wrong.
